eclipse-scout-core/src/tree/Tree.ts

```diff
@@ -1257,25 +1257,38 @@ export class Tree extends Widget implements TreeModel, Filterable<TreeNode> {
   }
 
   protected _updateMarkChildrenChecked(node: TreeNode) {
-    let treeNodeUpdate = this._checkParentsRecursive(node);
+    let checkResult = this._checkParentsRecursive(node);
 
     if (!node.initialized) {
       // No rendering of nodes which have not been initialized yet
-      treeNodeUpdate.removeNode(node);
+      checkResult.removeNode(node);
     }
 
     // Trigger events and render
-    this._renderNodesChecked(treeNodeUpdate.getNodesForRendering());
-    this.trigger('nodesChecked', {
-      nodes: treeNodeUpdate.getNodesForEventTrigger()
-    });
+    this._processTreeCheckNodesResult(checkResult);
   }
 
-  protected _renderNodesChecked(treeNodes: TreeNode[]) {
-    treeNodes.forEach(node => {
+  /**
+   * Processes a {@link TreeCheckNodesResult} object. It renders all updated nodes and triggers
+   * an {@code nodesChecked} event to send updated nodes to java.
```

Comment:
```suggestion
   * a `nodesChecked` event.
```

* an -> a
* The tree widget is not aware of the java part, it just emits the event and the TreeAdapter is responsible to forward that event to java.
* In TypeScript, we use backticks to format inline code in jsdoc